### PR TITLE
[Loading] update `z-index` to be above `TopBar`

### PR DESCRIPTION
**Pull request by yurm04**
⚠️ Remove changeset before merging ⚠️ 

### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-summer-editions/issues/996

This issue exists because there is a [new condition for summer editions to bump up the top bar z-index](https://github.com/Shopify/web/commit/203717f7cb2d080afd9743d069926c8ab9a614a3) when the `AppFrameTopBar` id is present. This moves the Topbar over the loading component which makes it disappear.

Not sure if this is the best place to make this fix now or if we should hack it in `web`, definitely not a good long term solution. Could use some input from @chloerice @gwyneplaine for more context

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/Frame/Frame.scss [hunk 1]
```diff
@@ -149,6 +149,11 @@ $sidebar-breakpoint: 1200px;
   width: 100%;
   height: $top-bar-height;
 
+  #{$se23} {
+    z-index: var(--p-z-index-10);
```
Comment (mattkubej) [inline polaris-react/src/components/Frame/Frame.scss:153]: Considering there is a patch already in web for the TopBar. I think we should consider extending the patch in web to fix this in the short term (i.e. before editions) or adjust Polaris to remove the patch that exists within web today and upgrade web with the change.

The use cases we need to be considerate of are any popovers, sidekick, settings, settings with sidekick, settings with search, bulk editor, and bulk editor with popovers. There are likely others, but these would be the major pieces we'd need to vet with adjusting the z-index.

Comment (yurm04) [reply polaris-react/src/components/Frame/Frame.scss:153]: @mattkubej let me know what you think about this approach in web https://github.com/Shopify/web/pull/99117

**Comment by yurm04** [comment]
closing in favor of https://github.com/Shopify/web/pull/99117

**Pull request closed without merging**